pkg/kopia/command/storage/secret_utils.go

```diff
@@ -183,6 +184,9 @@ func GetMapForLocationValues(
 	}
 	if locType != "" {
 		m[typeKey] = []byte(locType)
+		if locType == LocType(v1alpha1.LocationTypeS3Compliant) {
```

Comment:
```suggestion
		if loc, ok == LocType(v1alpha1.LocationTypeS3Compliant); ok && loc == locType {
```
Check if type assertion is successful to be on safer side. Otherwise it may panic at runtime of becomes incompatible in future

Reply:
Makes sense

Reply:
I checked but type conversion here does not return two values here.
Type assertion works only on interface I think. Also compilation itself would fail if type of either(LocType or LocationTypeS3Compliant) changes so we shouldn't have an issue.